### DISCUSSION TITLE
Is it possible to re-use unused IDs?

### DISCUSSION THREAD

**Discussion by stomko11** [zwave-js/zwave-js-ui#2036] [Q&A]
I have large zwave network, in some cases, I was not able to replace devices, thus ended up using exclusion, inclusion, etc. That, however, seems to take device ID, and even if device is properly excluded (or factory reset, does not matter really), I can't use the same ID again, and end up with gaps. So I have device IDs for instance 84, 85, 87, 89, ... 
There was a discussion if zwave netowork has limitation of actual IDs or just amount of nodes, I am still pretty far from there, but it would be nice if I could just take another free ID upon next inclusion. Is that even possible?

**Comment by robertsLando** [comment]
The only way would be to use replaceFailed node api but that requires your node to be dead and not removed. If it has been removed there is no way AFAIK to replace them

cc @AlCalzone

**Comment by AlCalzone** [comment]
If the ID is gone, meaning the node has been removed already, you can't reuse it easily. The ids wrap around after 232 (if the network isn't full) so you could (if you're really determined) include and exclude so often that you get back to the beginning.

You can reuse an ID by resetting the node while zwavejs2mqtt is shut down and then use the replace failed node function to replace it.
